### PR TITLE
Fix for RelayCompiler watch-mode

### PR DESCRIPTION
**Pull request by Yan-J**
Fixed the problem that RelayCompiler stops watching changes after an error.

### CODE REVIEW

**Comment by Yan-J** [comment]
@facebook-github-bot import

**Comment by wincent** [comment]
Awesome. Let's remove those `.DS_Store`, `packages/.DS_Store`, `packages/relay-compiler/.DS_Store` files that snuck in, and also undo the mode changes (the files which went from 644 to 755 executable).

File: packages/relay-compiler/codegen/RelayCodegenRunner.js [hunk 1]
```diff
@@ -247,14 +247,19 @@ class RelayCodegenRunner {
         this.parserWriters[parserName].forEach(writer =>
           dependentWriters.push(writer),
         );
-        if (!this.parsers[parserName]) {
-          // have to load the parser and make sure all of its dependents are set
-          await this.parseEverything(parserName);
-        } else {
-          this.parseFileChanges(parserName, files);
+        try{
```
Comment (wincent) [inline packages/relay-compiler/codegen/RelayCodegenRunner.js]: nit: space between `try` and `{`. Also, I don't know if GitHub is misrendering this but the indenting looks a bit off.

File: packages/relay-compiler/codegen/RelayCodegenRunner.js [hunk 2]
```diff
@@ -247,14 +247,19 @@ class RelayCodegenRunner {
...
+            // have to load the parser and make sure all of its dependents are set
+            await this.parseEverything(parserName);
+          } else {
+            this.parseFileChanges(parserName, files);
+          }
+          await Promise.all(dependentWriters.map(writer => this.write(writer)));
         }
-
-        await Promise.all(dependentWriters.map(writer => this.write(writer)));
+        catch (error){
```
Comment (wincent) [inline packages/relay-compiler/codegen/RelayCodegenRunner.js]: nit: space between `)` and `{`

File: packages/relay-compiler/core/ASTConvert.js [hunk 3]
```diff
@@ -125,10 +125,14 @@ function convertASTDefinitions(
     definitions: (operationDefinitions: Array<$FlowFixMe>),
   };
   // Will throw an error if there are validation issues
-  RelayValidator.validate(validationAST, schema, validationRules);
-  return operationDefinitions.map(definition =>
-    RelayParser.transform(schema, definition),
-  );
+  try{
```
Comment (wincent) [inline packages/relay-compiler/core/ASTConvert.js]: nit: space between`try` and `{`

File: packages/relay-compiler/core/ASTConvert.js [hunk 4]
```diff
@@ -125,10 +125,14 @@ function convertASTDefinitions(
...
+    RelayValidator.validate(validationAST, schema, validationRules);
+    return operationDefinitions.map(definition =>
+      RelayParser.transform(schema, definition),
+    );
+  }catch(error){
```
Comment (wincent) [inline packages/relay-compiler/core/ASTConvert.js]: nit: `} catch (error) {` (spaces)

File: packages/relay-compiler/core/FileParser.js [hunk 5]
```diff
@@ -42,7 +42,16 @@ class FileParser {
     let documents = ImmutableMap();
 
     files.forEach(file => {
-      const doc = this._parse(path.join(this._baseDir, file));
+      //const doc = this._parse(path.join(this._baseDir, file));
```
Comment (wincent) [inline packages/relay-compiler/core/FileParser.js]: Let's remove this comment.

**Comment by wincent** [comment]
This is so good; fixes probably the most annoying bug with the compiler that many people have run in to!

**Comment by wincent** [comment]
If you want you can add something like this to make sure the related issue gets closed automatically when you land this:

> Closes: https://github.com/facebook/relay/issues/1765

**Comment by nodkz** [comment]
Does `relay-compiler` track changes in schema.graphql file in watch mode with regenerating all __generated__ files? 

Related: https://github.com/nodkz/babel-plugin-transform-relay-hot/issues/6

**Comment by cbranch101** [comment]
Any idea when this might land in the `relay-compiler` npm package, or what the proper channel is to figure that out?

**Pull request closed without merging**